deepspeed/runtime/engine.py

```diff
@@ -2289,7 +2299,8 @@ def load_moe_state_dict(checkpoint_path,
                             old_moe_load,
                             model=None,
                             mpu=None,
-                            num_experts=1):
+                            num_experts=1,
+                            checkpoint_engine=CheckpointEngine()):
```

Comment:
Why not pass self.checkpoint_engine instead of CheckpointEngine()?

Reply:
It is class method where we can not get the variables from instance.
```python
    @staticmethod
    def load_moe_state_dict(checkpoint_path,
                            tag,
                            state_dict,
                            old_moe_load,
                            model=None,
                            mpu=None,
                            num_experts=1,
                            checkpoint_engine=CheckpointEngine()):
```

Reply:
Fair point. But right now, this will default to torch checkpoint engine, correct?

Reply:
Yep~